### ISSUE TITLE
Helm 2.9.0 is giving "connection refused" when deploying charts

### ISSUE THREAD

**Issue by arun-gupta** [kubernetes/helm#4001]
Following up from https://github.com/kubernetes/helm/issues/2520. Filing as a new issue as this is occurring with 2.9.0 only and does not occur with 2.8.1.

```
$ helm version
Client: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.9.0", GitCommit:"f6025bb9ee7daf9fee0026541c90a6f557a3e0bc", GitTreeState:"clean"}
```

Can't even delete Helm:

```
$ helm reset
Error: Get http://localhost:8080/api/v1/namespaces/kube-system/configmaps?labelSelector=OWNER%!D(MISSING)TILLER: dial tcp [::1]:8080: connect: connection refused
```

Had to delete as:

```
$ kubectl delete deployment tiller-deploy --namespace kube-system
deployment.extensions "tiller-deploy" deleted
```

Switched to 2.8.1:

```
$ brew switch kubernetes-helm 2.8.1
Cleaning /usr/local/Cellar/kubernetes-helm/2.6.2
Cleaning /usr/local/Cellar/kubernetes-helm/2.8.1
Cleaning /usr/local/Cellar/kubernetes-helm/2.9.0
46 links created for /usr/local/Cellar/kubernetes-helm/2.8.1
```

Kubectl version:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.1", GitCommit:"d4ab47518836c750f9949b9e0d387f20fb92260b", GitTreeState:"clean", BuildDate:"2018-04-13T22:29:03Z", GoVersion:"go1.9.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-07T11:55:20Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```

Cluster created using kops on AWS.

More details at https://github.com/arun-gupta/kubernetes-chaos/issues/2

**Comment by bacongobbler** [comment]
Hey there! This issue has been reported several times at this point. Please see #3990 and the related (duplicate) tickets in that thread. There are several described worarounds, and a patch has been applied to the canaries. A v2.9.1 with that patch will follow once the core team has recovered from jet lag ;)

**Comment by bacongobbler** [comment]
For reference, the workaround is here: https://github.com/kubernetes/helm/issues/3985#issuecomment-385105880

**Comment by bhalothia** [comment]
This fixed it for me.

`kubectl -n kube-system patch deployment tiller-deploy -p '{"spec": {"template": {"spec": {"automountServiceAccountToken": true}}}}'`

**Comment by bacongobbler** [comment]
This has been fixed in helm v2.9.1, which is available here: https://github.com/kubernetes/helm/releases/v2.9.1

Thanks everyone!
